### PR TITLE
allow ignore sides and start using our logger so we see more colors

### PR DESCRIPTION
**Pull request by shedaniel**

### CODE REVIEW

**Comment by shedaniel** [comment]
also fixed your checkstyle violation

File: src/main/java/com/patchworkmc/Patchwork.java [hunk 1]
```diff
@@ -94,24 +103,24 @@ public static void main(String[] args) throws Exception {
...
-			System.out.println("=== Transforming " + modName + " ===");
+			LOGGER.info("=== Transforming " + modName + " ===");
 
 			try {
 				transformMod(currentPath, file, currentPath.resolve("output"), modName, mappings, intermediaryMappings);
 			} catch (Exception e) {
-				System.err.println("Transformation failed, going on to next mod: ");
+				LOGGER.error("Transformation failed, going on to next mod: ");
 
 				e.printStackTrace();
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/Patchwork.java:132]: Probably want to make all the `e.printStackTrace()` lines go through the logger

**Comment by shedaniel** [comment]
oh no merge conflicts

**Review by coderbot16 — APPROVED** [review]
LGTM

**Comment by shedaniel** [comment]
level trace only outputs to console, it is a great way to debug without messing up with the user UI.

this pr is ready for merge, I will start making another pr to try to do the command line.

File: src/main/java/com/patchworkmc/event/generator/SubscribeEventGenerator.java [hunk 2]
```diff
@@ -23,6 +24,8 @@ public static String generate(String targetClass, SubscribeEvent entry, ClassVis
 		// Add a default constructor
 
 		if (instance) {
+			Patchwork.LOGGER.trace("Instance subscribe event on " + targetClass + " (method: " + entry.getMethod() + ")");
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/event/generator/SubscribeEventGenerator.java]: This was removed

File: src/main/java/com/patchworkmc/patch/ItemGroupTransformer.java [hunk 3]
```diff
@@ -49,12 +52,14 @@ public void visitMethodInsn(int opcode, String owner, String name, String descri
 				super.visitMethodInsn(opcode, owner, name, descriptor, isInterface);
 			}
 
+			Patchwork.LOGGER.trace("Found InvokeSpecial in <init> " + owner + " " + name + " " + descriptor + " " + isInterface);
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/patch/ItemGroupTransformer.java]: This was removed!

File: src/main/java/com/patchworkmc/patch/ItemGroupTransformer.java [hunk 4]
```diff
@@ -32,6 +34,7 @@ public void visit(int version, int access, String name, String signature, String
 	@Override
 	public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
 		if (applies && name.equals("<init>")) {
+			Patchwork.LOGGER.trace("Patching an ItemGroup to use Patchwork");
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/patch/ItemGroupTransformer.java]: This was removed!

**Review by coderbot16 — COMMENTED** [review]
Overall looks good, just a few added logspam messages that I removed.

File: src/main/java/com/patchworkmc/event/generator/SubscribeEventGenerator.java [hunk 5]
```diff
@@ -4,6 +4,7 @@
 import org.objectweb.asm.MethodVisitor;
 import org.objectweb.asm.Opcodes;
 
+import com.patchworkmc.Patchwork;
```
Comment (coderbot16) [inline src/main/java/com/patchworkmc/event/generator/SubscribeEventGenerator.java]: unused import

**Review by coderbot16 — COMMENTED** [review]
remove unused imports pls

**Review by coderbot16 — APPROVED** [review]
Thanks!

**Pull request merged**